includes/publisher/js/src/components/RelationshipModal.jsx

```diff
@@ -0,0 +1,297 @@
+/* global atlasContentModelerFormEditingExperience */
+import React, { useContext, useEffect, useState } from "react";
+import { sprintf, __ } from "@wordpress/i18n";
+import Modal from "react-modal";
+const { wp } = window;
+const { date, apiFetch } = wp;
+
+Modal.setAppElement("#atlas-content-modeler-fields-app");
+
+/**
+ * The modal component for editing a relationship.
+ *
+ * @param {Object} field The relationship field.
+ * @param {Boolean} isOpen Whether or not the model is open.
+ * @param {Function} setIsOpen - Callback for opening and closing modal.
+ * @returns {JSX.Element} Modal
+ */
+export default function RelationshipModal({ field, isOpen, setIsOpen }) {
+	const [page, setPage] = useState(1);
+	const [pagedEntries, setPagedEntries] = useState({});
+	const [totalEntries, setTotalEntries] = useState(0);
+	const [selectedEntry, setSelectedEntry] = useState(); // TODO: set initial state value from stored field value.
```

Comment:
```suggestion
	const [selectedEntry, setSelectedEntry] = useState(parseInt(field?.value));
```

This ensures that the currently selected entry will have its radio button active when you load an existing post and click “link new reference”.